File: tests/fixtures/python_bug/buggy.py
"""Buggy Python script — NameError: undefined variable."""


def calculate_ratio(numerator, denominator):
    if denominator == 0:
        return 0
    return numerator / denom  # BUG: typo — should be 'denominator'


def main():
    result = calculate_ratio(10, 2)
    print(f"Result: {result}")


if __name__ == "__main__":
    main()

File: tests/fixtures/js_bug/buggy.js
// Buggy JavaScript — ReferenceError: undefined variable
function calculateRatio(numerator, denominator) {
    if (denominator === 0) {
        return 0;
    }
    return numerator / denom;  // BUG: 'denom' is undefined, should be 'denominator'
}

function main() {
    const result = calculateRatio(10, 2);
    console.log(`Result: ${result}`);
}

main();
